tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using MultiFunctionApplicationHelpers;
+using NewRelic.Agent.IntegrationTestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NewRelic.Agent.IntegrationTests.InfiniteTracing
+{
+    public abstract class InfiniteTracingTestsBase<TFixture> : NewRelicIntegrationTest<TFixture>
+        where TFixture : ConsoleDynamicMethodFixture
+    {
+        private readonly TFixture _fixture;
+
+        public InfiniteTracingTestsBase(TFixture fixture, ITestOutputHelper output):base(fixture)
+        {
+            _fixture = fixture;
+            _fixture.SetTimeout(System.TimeSpan.FromMinutes(2));
+            _fixture.TestLogger = output;
+
+            _fixture.AddCommand($"InfiniteTracingTester StartAgent");
+            _fixture.AddCommand($"InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand($"InfiniteTracingTester Wait");
+
+
+            _fixture.Actions
+            (
+                setupConfiguration: () =>
+                {
+                    var configModifier = new NewRelicConfigModifier(fixture.DestinationNewRelicConfigFilePath);
+
+                    configModifier.ForceTransactionTraces()
+                    .EnableDistributedTrace()
+                    .EnableInfinteTracing(_fixture.TestConfiguration.TraceObserverUrl)
+                    .SetLogLevel("finest");
+                }
+            );
+
+            _fixture.Initialize();
+        }
+
+        [Fact]
+        public void Test()
+        {
+            var expectedSeenCount = 2;
```

Comment:
Given that the way the test application is being exercised (one method that creates a transaction is being called only once), what is causing the expected count of span(s) seen, sent and received to be 2?

Reply:
https://github.com/newrelic/newrelic-dotnet-agent/blob/5b375b3c38517c8326aac0a70a4901f99f1a483c/src/Agent/NewRelic/Agent/Core/Transformers/TransactionTransformer/TransactionTransformer.cs#L322. According to this logic, we always send an extra count. I believe 1 span is the instrumented method, the extra is for root span.

Reply:
Cool, thanks.  I think we should add a comment to the test to explain this to the next engineer who looks at it.

Reply:
Added.